simple/unexpected_msg.c

```diff
@@ -57,31 +57,56 @@ struct fi_context *tx_ctxs;
 struct fi_context *rx_ctxs;
 static bool send_data = false;
 char *tx_bufs, *rx_bufs;
-
+static struct fid_mr *tx_mr, *rx_mr;
+static void *tx_mr_desc, *rx_mr_desc;
+static size_t tx_len, rx_len;
 
 static int alloc_bufs(void)
 {
-	tx_bufs = calloc(concurrent_msgs, opts.transfer_size);
-	rx_bufs = calloc(concurrent_msgs, opts.transfer_size);
+	int rc;
+
+	tx_len = opts.transfer_size + ft_tx_prefix_size();
+	rx_len = opts.transfer_size + ft_rx_prefix_size();
+	tx_bufs = calloc(concurrent_msgs, tx_len);
+	rx_bufs = calloc(concurrent_msgs, rx_len);
 	tx_ctxs = malloc(sizeof(*tx_ctxs) * concurrent_msgs);
 	rx_ctxs = malloc(sizeof(*rx_ctxs) * concurrent_msgs);
 	if (!tx_bufs || !rx_bufs || !tx_ctxs || !rx_ctxs)
 		return -FI_ENOMEM;
+	if (fi->domain_attr->mr_mode & FI_MR_LOCAL) {
+		rc = fi_mr_reg(domain, tx_bufs, concurrent_msgs * tx_len,
+			       FI_SEND, 0, FT_MR_KEY + 1, 0, &tx_mr, NULL);
+		if (rc < 0)
+			return rc;
+		tx_mr_desc = fi_mr_desc(tx_mr);
+		rc = fi_mr_reg(domain, rx_bufs, concurrent_msgs * rx_len,
+			       FI_RECV, 0, FT_MR_KEY + 2, 0, &rx_mr, NULL);
+		if (rc < 0)
```

Comment:
nit: here and above check if(rc) -- any non-zero return value is unexpected and should be flagged as an error.